src/os/bluestore/BlueStore.cc

```diff
@@ -5354,13 +5354,27 @@ int BlueStore::_balance_bluefs_freespace(PExtentVector *extents)
     dout(10) << __func__ << " gifting " << gift
 	     << " (" << byte_u_t(gift) << ")" << dendl;
 
-    int64_t alloc_len = alloc->allocate(gift,
-					cct->_conf->bluefs_shared_alloc_size,
+    // first try to allocate larger chunks (usually the bluefs alloc
```

Comment:
are we going to have this in master/nautilus?

Reply:
master/nautlius doesn't have any allocation here.. relies entirely on fallback it seems

Reply:
@liewegas - I mean whether we want to use the same two-step approach when doing on-demand bluefs space allocation in master.